lumicks/pylake/detail/confocal.py

```diff
@@ -273,13 +273,39 @@ def _to_spatial(self, data):
         raise NotImplementedError
 
     @cachetools.cachedmethod(lambda self: self._cache)
-    def _image(self, channel):
-        assert channel in ("red", "green", "blue")
+    def _image(self, channel) -> np.ndarray:
+        """Returns a reconstructed image.
+
+        Note: Do *not* modify or return this image without making a copy, as it will return a
+        reference to the cached image, which makes the `ConfocalImage` mutable.
+
+        Parameters
+        ----------
+        channel : str
+            Channel to return. Must be "red", "green" or "blue".
+        """
+        if channel not in ("red", "green", "blue"):
+            raise ValueError(f'Channel must be "red", "green" or "blue", got "{channel}".')
+
         return self._image_factory(self, channel)
 
     @cachetools.cachedmethod(lambda self: self._cache)
-    def _timestamps(self, channel, reduce=timestamp_mean):
-        assert channel == "timestamps"
+    def _timestamps(self, channel, reduce=timestamp_mean) -> np.ndarray:
+        """Returns reconstructed timestamps.
+
+        Note: Do *not* modify or return this image without making a copy, as it will return a
+        reference to the cached image, which makes the `ConfocalImage` scan mutable.
```

Comment:
```suggestion
        Warning: Do *not* modify or return this image without making a copy, as it will return a
        reference to the cached image, which makes the `ConfocalImage` scan mutable.
```